### ISSUE TITLE
Display who is currently participating in the demo page

### ISSUE THREAD

**Issue by hackerwins** [hackerwins/homepage-old#21]
The editors on the [demo page](https://yorkie.dev/demo) are more valuable when editing by people at the same time. We recently added Peer-Awareness feature that lets users know who has concurrently connected. It would be nice to display who is currently participating in the demo page using the feature. I think it's also good to display network status.

<img width="908" alt="Screen Shot 2020-09-23 at 8 37 47 AM" src="https://user-images.githubusercontent.com/2059311/93947971-1d6ff200-fd78-11ea-8e80-88458aba480e.png">

 - https://github.com/yorkie-team/yorkie-js-sdk/issues/66
 - https://github.com/yorkie-team/yorkie/issues/48

**Comment by lqez** [comment]
Is anyone working on this issue?

**Comment by hackerwins** [comment]
@lqez No one is participating yet. Are you willing to participate in this issue?

**Comment by hannapa1114** [comment]
hi, can I work on this issue?

**Comment by hackerwins** [comment]
@hannapa1114 Yes. Thank you for your interest in contributing.

**Comment by lqez** [comment]
@hackerwins I've tried to add this feature on the demo page, but I'm not sure how to handle `peers-changed` events properly.

I saw a similar implementation at https://github.com/yorkie-team/codepair/blob/3cd53bd46fb7772b1b262ea3f74fac3ae18d1f0f/webapp/src/features/peerSlices.ts#L40 but still confusing.

It looks like it sets all status of peers as `Disconnected` not in `changedPeers` first and regards all peers in  `changedPeers` as `Connected` status again, right?

I copied this behavior but got the wrong result.
```
61bce44507637f0e8ba1a057: {id: '61bce44507637f0e8ba1a057', status: 'connected', metadata: {…}}
61bce44907637f0e8ba1a05e: {id: '61bce44907637f0e8ba1a05e', status: 'disconnected', metadata: {…}}
61bce46907637f0e8ba1a09a: {id: '61bce46907637f0e8ba1a09a', status: 'disconnected', metadata: {…}}
```
3 clients are should be displayed as `connected`. And too many events occurred for a simple entrance situation. (almost 30+ events)

I think there must be a `status` data on `peers-changed` events. 

You can find out my code at https://github.com/lqez/yorkie-team.github.io/blob/add-peer-awareness/static/js/demo-peer-awareness.js

**Comment by hackerwins** [comment]
@lqez Thanks for the report.

One of the reasons ` peers-changed` event occurs multiple times is that each time `attachDocument` is executed, the previously established stream is disconnected and reconnected(In the demo, 6 documents are attached, so the stream is reconnected 6 times). We are forgetting that we have to deal with this issue.

I think the problem of the disconnected clients is because the order in which the events are delivered is not guaranteed.

It's good to start with keeping the stream in `attachDocument` then checking the last problem.

**Comment by lqez** [comment]
Issue update:
This problem only occurs when a client tried to attach multiple documents.

Yorkie JS SDK receives `peerChanged` event from Yorkie Agent and also maintains peer list with payloads from the event. Payloads from Agent only contain 'changed' client information.

JS SDK also provides `peer-changed` events to JS client but its payload is different from Agent's. It's a whole list of peers, not just 'changed' peers.

This causes confusion and makes developers hard to understand how to handle peer-awareness events properly. (even CodePair code handle it in a weird way)

**Comment by hackerwins** [comment]
I replace the multiple documents on the example page with one document: 3e68494f2eae80fa33bdf41ec53b0f698ecc40ce

@lqez You can resolve this issue without waiting for the Yorkie update.

**Comment by hackerwins** [comment]
This seems to be fixed by #31
